redis_elasticache_client.go

```diff
@@ -36,7 +39,17 @@ func (r *redisElastiCacheDB) Initialize(_ context.Context, req dbplugin.Initiali
 		return dbplugin.InitializeResponse{}, err
 	}
 
-	creds, err := awsutil.RetrieveCreds(r.config.Username, r.config.Password, "", r.logger)
+	// If primary connection attributes are not set, try to fall back on the deprecated values for backward compatibility
+	accessKey := r.config.AccessKeyID
+	if accessKey == "" && r.config.Username != "" {
+		accessKey = r.config.Username
+	}
+	secretKey := r.config.SecretAccessKey
+	if secretKey == "" && r.config.SecretAccessKey != "" {
+		secretKey = r.config.SecretAccessKey
+	}
+
+	creds, err := awsutil.RetrieveCreds(accessKey, secretKey, r.config.SessionToken, r.logger)
```

Comment:
Why are we setting the SessionToken when previously we were not?

Reply:
It lets users use STS credentials and it seems to be a common AWS authentication option we front on AWS plugins that require an access key id & secret access key, like the [Vault Agent].(https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/aws)

Temporary creds might be ill-suited for a plugin though? Maybe we better not make it available unless a user asks for it and can provide a legitimate use case where managing STS creds is necessary.

Reply:
I agree with that. Thanks!